### PR TITLE
modified CardContainerView#updateAlpha.

### PR DESCRIPTION
**Pull request by katsuya-yamada-eureka**
[matter]
・top card swipe to right-bottom(roughly 60 degrees)
・end swipe
　-> At this time,
          applied overlay : Bottom
          applied Action up processing : Right

[issue]
・check SwipingDirection when overlay processing( CardContainerView#updateAlpha)
    HORIZONTAL :  watch only X-axis
    VERTICAL : watch only Y-axis
　 FREEDOM_NO_BOTTOM : watch XY-axis, and consider movement of x,y (ignore bottom)
　 FREEDOM : watch XY-axis, and consider movement of x,y

### CODE REVIEW

**Comment by luongvo** [comment]
@katsuya-yamada-eureka could you explain more about the issue and your idea?

**Comment by katsuya-yamada-eureka** [comment]
@luongvo 
I'm sorry, was fixed! :)

**Comment by futabooo** [comment]
👍

File: cardstackview/src/main/java/com/yuyakaido/android/cardstackview/internal/CardContainerView.java [hunk 1]
```diff
@@ -215,23 +215,44 @@ private void updateAlpha() {
...
+                setOverlayAlpha(Math.abs(percentY));
             } else {
-                showRightOverlay();
+                showHorizontalOverlay(percentX);
             }
-            setOverlayAlpha(Math.abs(percentX));
-        }else{
-            if (percentY < 0) {
-                showTopOverlay();
+        } else if (option.swipeDirection == SwipeDirection.FREEDOM) {
```
Comment (yuyakaido) [inline cardstackview/src/main/java/com/yuyakaido/android/cardstackview/internal/CardContainerView.java]: If we call `setSwipeDirection(Arrays.asList(SwipeDirection.Left))`, the overlay does not work...
We should add `else clause` to handle all of swipe direction patterns.

So, add this code as else clause.
```
        else {
            if (Math.abs(percentX) > Math.abs(percentY)){
                if (percentX < 0) {
                    showLeftOverlay();
                } else {
                    showRightOverlay();
                }
                setOverlayAlpha(Math.abs(percentX));
            } else {
                if (percentY < 0) {
                    showTopOverlay();
                } else {
                    showBottomOverlay();
                }
                setOverlayAlpha(Math.abs(percentY));
            }
        }
```

Comment (yuyakaido) [reply cardstackview/src/main/java/com/yuyakaido/android/cardstackview/internal/CardContainerView.java]: I will fix this problem in https://github.com/yuyakaido/CardStackView/issues/88

**Pull request merged**